File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    data.sort(key=lambda x: (x[col - 1], -x[0]))

    hash_value = 0

    for i in range(row_begin - 1, row_end):
        si = sum(data[i][j] % (i + 1) for j in range(len(data[i])))
        hash_value ^= si

    return hash_value

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.java
import java.util.Arrays;

class Solution {
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        Arrays.sort(data, (a, b) -> {
            if (a[col - 1] != b[col - 1]) {
                return Integer.compare(a[col - 1], b[col - 1]);
            } else {
                return Integer.compare(b[0], a[0]);
            }
        });
        
        int hashValue = 0;
        for (int i = row_begin - 1; i < row_end; i++) {
            int si = 0;
            for (int j = 0; j < data[i].length; j++) {
                si += data[i][j] % (i + 1);
            }
            hashValue ^= si;
        }
        
        return hashValue;
        
    }
}